assets/js/src/public.js

```diff
@@ -60,11 +60,22 @@ jQuery(function ($) { // eslint-disable-line func-names
           data: formData.data,
         })
           .fail(function handleFailedPost(response) {
-            form.find('.mailpoet_validate_error').html(
-              response.errors.map(function buildErrorMessage(error) {
-                return error.message;
-              }).join('<br />')
-            ).show();
+            if (
+              response.meta !== undefined
+                && response.meta.redirect_url !== undefined
+            ) {
+              // go to page
+              window.top.location.href = response.meta.redirect_url;
+            } else {
+              if (response.meta && response.meta.refresh_captcha) {
+                $('.mailpoet_captcha_update').click();
```

Comment:
What about creating `updateCaptcha` function and call it directly here and also within `$('.mailpoet_captcha_update').click()`. In my opinion it would be more robust and readable than triggering it using dom event.